### ISSUE TITLE
KeyError in ProfilingPanel after upgrade to 1.4 

### ISSUE THREAD

**Issue by pendolf** [jazzband/django-debug-toolbar#792]
Hello! After upgrade django-debug-toolbar from 1.3.2 to 1.4 debug_toolbar.panels.profiling.ProfilingPanel stop working.
Traceback: http://dpaste.com/347D0XH

**Comment by jhg-tgu** [comment]
```python
Traceback:
File "/opt/env/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  223.                 response = middleware_method(request, response)
File "/opt/env/lib/python3.4/site-packages/debug_toolbar/middleware.py" in process_response
  129.                 panel.generate_stats(request, response)
File "/opt/env/lib/python3.4/site-packages/debug_toolbar/panels/profiling.py" in generate_stats
  170.         root = FunctionCall(self.stats, self.stats.get_root_func(), depth=0)
File "/opt/env/lib/python3.4/site-packages/debug_toolbar/panels/profiling.py" in __init__
  51.             self.stats = statobj.stats[func][:4]```

**Comment by jhg-tgu** [comment]
@robhudson @jezdez @dcramer when estimate this will be fixed?

**Comment by aaugustin** [comment]
Most likely when someone submits a good patch.

**Comment by blueyed** [comment]
Any more information about the error?

I am seeing it at `/admin/login` only?!

**Comment by d9pouces** [comment]
More precisely, I only encounter this problem in /admin/login when I'm not logged in.
The DjangoDebugToolbarStats class does not find the root function.

**Comment by jhg-tgu** [comment]
I find this in all URLs with and without login.

**Comment by sasha0** [comment]
The issue exists for decorated view functions (for instance, `/admin/login`), which `ProfilingPanel` can't find root function for and hence to build hierarchy of function calls.

**Comment by sasha0** [comment]
Did more investigation and found out that issue exists when decorated view has calls of decorated view in it, more than that the case exists when they wrapped with the same decorator. 
I did another tests and got the same result for this case. 

Eventually, we can't determine root function and hence, build calls tree. 
Continue to investigate how to determine root function, at the meantime as a temporary solution we could disable `ProfilePanel` for this cases which at least will prevent particular page within other Django Debug Toolbar panels from being broken.

**Comment by tony** [comment]
Started getting this randomly after:

1. being logged into admin with a session
2. loaddata on user information
3. runserver_plus restarts
4. admin wants to go to login screen

then I get the KeyError issue from profiling

if I disable the ProfilingPanel and log back into admin, everything is fixed

**Comment by radwon** [comment]
Having encountered this error myself, and after some debugging I've found the problem to be caused by self.stats.get_root_func() returning None within the generate_stats() method of the ProfilingPanel class.

To fix the issue, I added a simple check for the existence of a root function before proceeding. It seems to have solved this issue, so I've submitted a pull request.

Hope this works for everyone else.
